File: app/src/main/java/com/frankly/randomdog/model/DogModel.java
package com.frankly.randomdog.model;

/**
 * @author: ArshVardhanAtreya <arshvardhan@yahoo.co.in>
 */


public class DogModel {

    private String imageURL;
    private String key;

    public DogModel(String imageURL) {
        this.imageURL = imageURL;
        this.key = imageURL.substring(imageURL.lastIndexOf("/") + 1);
    }

    public String getImageURL() {
        return imageURL;
    }

    public String getKey() {
        return key;
    }

}

File: app/src/main/java/com/frankly/randomdog/model/DogModel.kt
package com.frankly.randomdog.model


/**
 * @author: ArshVardhanAtreya <arshvardhan></arshvardhan>@yahoo.co.in>
 */
class DogModel(val dogImageURL: String?) {
    val dogImageKey: String? = dogImageURL?.substring(dogImageURL.lastIndexOf("/") + 1)
    fun getImageURL(): String? {
        return dogImageURL
    }

    fun getKey(): String? {
        return dogImageKey
    }

}